src/es/templescanesp/src/eu/kanade/tachiyomi/extension/es/templescanesp/TempleScanEsp.kt

```diff
@@ -31,6 +33,20 @@ class TempleScanEsp :
         else -> preferences.prefBaseUrl
     }
 
+    private val fetchedDomainUrl: String by lazy {
+        if (!preferences.fetchDomainPref()) return@lazy preferences.prefBaseUrl
+        try {
+            val initClient = network.cloudflareClient
+            val headers = super.headersBuilder().build()
+            val host = initClient.newCall(GET(preferences.prefBaseUrl, headers)).execute().request.url.host
```

Comment:
set `followRedirect` to false for the client, and get new url from `Location` header (3xx response)